ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -158,18 +174,38 @@ def create_input_processors(
         :param unnormalized_inputs: Vector inputs that should not be normalized, and added to the vector
             obs.
         :param normalize: Normalize all vector inputs.
-        :return: Tuple of visual encoders and vector encoders each as a list.
+        :return: Tuple of :
+         - ModuleList of the encoders (None if the input requires to be processed with a variable length
+         observation encoder)
+         - A list of embedding sizes (0 if the input requires to be processed with a variable length
+         observation encoder)
+         - A list of the inputs that need to be processed by a variable length observation encoder.
         """
         encoders: List[nn.Module] = []
+        var_encoders: List[nn.Module] = []
         embedding_sizes: List[int] = []
-        for obs_spec in observation_specs:
+        var_len_indices: List[int] = []
+        for idx, obs_spec in enumerate(observation_specs):
             encoder, embedding_size = ModelUtils.get_encoder_for_obs(
-                obs_spec.shape, normalize, h_size, vis_encode_type
+                obs_spec, normalize, h_size, vis_encode_type
             )
             encoders.append(encoder)
             embedding_sizes.append(embedding_size)
+            if encoder is None:
+                var_len_indices.append(idx)
 
-        return (nn.ModuleList(encoders), embedding_sizes)
+        x_self_size = sum(embedding_sizes)  # The size of the "self" embedding
+        for idx in var_len_indices:
+            var_encoders.append(
+                EntityEmbedding(
+                    x_self_size,
+                    observation_specs[idx].shape[1],
+                    observation_specs[idx].shape[0],
+                    h_size,
+                    concat_self=True,
```

Comment:
This is defaulted to True

Reply:
I still think having it explicit is good for the reader, but I can remove

Reply:
Should we just remove the default arg and force people to choose?

Reply:
Isn't a concat_self = True just a linear encoder 🤔 
I think I am okay removing the default

Reply:
concat_self=False is a linear encoder yes